modules/jackson-backend/src/main/scala/tethys/jackson/JacksonTokenIterator.scala

```diff
@@ -1,12 +1,19 @@
 package tethys.jackson
 
-import com.fasterxml.jackson.core.{JsonParser, JsonTokenId}
+import com.fasterxml.jackson.core.{JsonParseException, JsonParser, JsonTokenId}
 import tethys.commons.Token
 import tethys.commons.Token._
 import tethys.readers.tokens.{BaseTokenIterator, TokenIterator}
 
 class JacksonTokenIterator(jsonParser: JsonParser) extends BaseTokenIterator {
-  private[this] var token: Token = fromId(jsonParser.currentTokenId())
+  private[this] var token: Token = {
+    try {
+      jsonParser.nextToken() // move parser to first token
+      fromId(jsonParser.currentTokenId())
+    } catch {
+      case _: JsonParseException => UnrecognizedStart
```

Comment:
I think that is a wrong behavior.
JsonParseException could be thrown in any position, but it's handled only here
And new type of token :(
Maybe it's better to fix this issue in `tethys.ReaderReaderOps` functions?

Reply:
Another option is to change signature of `tethys.readers.tokens.TokenIteratorProducer#fromReader` from `Reader => TokenIterator` to `Reader => Either[ReaderError, TokenIterator]`